### PR TITLE
storage: Allow creation of swap and maintain fstab entries

### PR DESCRIPTION
**Pull request by mvollmer**
---

### Storage: improved support for swap

Cockpit can now format block devices as swap and will maintain a fstab entry for them.

### CODE REVIEW

**Comment by mvollmer** [comment]
@marusak For excellent swap support also in the normal Cockpit mode, we would need to write entries for swap devices into fstab, maintain their "noauto" option depending on whether the device is active, do some "misswapping" warnings, and remove the entry during teardown. That's entirely possible, but might not be worth the effort. Opinions?

**Comment by jelly** [comment]
> @marusak For excellent swap support also in the normal Cockpit mode, we would need to write entries for swap devices into fstab, maintain their "noauto" option depending on whether the device is active, do some "misswapping" warnings, and remove the entry during teardown. That's entirely possible, but might not be worth the effort. Opinions?

In my own opinion I would like this to be fully supported in Cockpit (even if it is dying tech most distro's nowadays only use zram/zswap). Keeping the amount of custom Anaconda features low would a good thing to strive for in my opinion.

File: pkg/storaged/block/format-dialog.jsx [hunk 1]
```diff
@@ -438,6 +439,10 @@ function format_dialog_internal(client, path, start, size, enable_dos_extended,
                     partition_type = "21686148-6449-6e6f-744e-656564454649";
                 }
 
+                if (type == "swap") {
+                    partition_type = block_ptable.Type == "dos" ? "0x82" : "0657fd6d-a4ab-43c4-84e5-0933c84b4f4f";
```
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:443]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240115-124335-f90ef176-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by mvollmer** [comment]
> In my own opinion I would like this to be fully supported in Cockpit (even if it is dying tech most distro's nowadays only use zram/zswap).

Yes, that's the right thing to do(tm), let's do it.

File: pkg/storaged/block/format-dialog.jsx [hunk 2]
```diff
@@ -423,6 +429,8 @@ function format_dialog_internal(client, path, start, size, enable_dos_extended,
             else if (trigger == "type") {
                 if (dlg.get_value("type") == "empty") {
                     dlg.update_actions({ Variants: action_variants_for_empty });
+                } else if (dlg.get_value("type") == "swap") {
```
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:432]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240116-090759-3ab49c70-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx [hunk 3]
```diff
@@ -450,6 +458,10 @@
                     partition_type = "21686148-6449-6e6f-744e-656564454649";
                 }
 
+                if (type == "swap") {
+                    partition_type = block_ptable.Type == "dos" ? "0x82" : "0657fd6d-a4ab-43c4-84e5-0933c84b4f4f";
```
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:462]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240116-090759-3ab49c70-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx [hunk 4]
```diff
@@ -532,6 +544,17 @@
                     }
                 }
 
+                if (type == "swap") {
+                    config_items.push(["fstab", {
+                        dir: { t: 'ay', v: encode_filename("none") },
+                        type: { t: 'ay', v: encode_filename("swap") },
+                        opts: { t: 'ay', v: encode_filename(mount_now ? "defaults" : "noauto") },
```
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:551]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240116-090759-3ab49c70-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx [hunk 5]
```diff
@@ -597,7 +620,10 @@
                         return (client.wait_for(() => block_fsys_for_block(path))
                                 .then(block_fsys => client.mount_at(client.blocks[block_fsys.path],
                                                                     mount_point)));
-                    if (is_encrypted(vals) && is_filesystem(vals) && !mount_now)
+                    if (type == "swap" && mount_now)
+                        return (client.wait_for(() => client.blocks_swap[path])
+                                .then(block_swap => block_swap.Start({})));
+                    if (is_encrypted(vals) && (is_filesystem(vals) || type == "swap") && !mount_now)
```
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:626]: These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240116-090759-3ab49c70-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/swap/swap.jsx [hunk 6]
```diff
@@ -27,14 +27,63 @@ import { useEvent } from "hooks";
...
+async function set_swap_noauto(block, noauto) {
+    for (const conf of block.Configuration) {
+        if (conf[0] == "fstab") {
+            const options = parse_options(decode_filename(conf[1].opts.v));
+            extract_option(options, "defaults");
+            extract_option(options, "noauto");
+            if (noauto)
+                options.push("noauto");
+            if (options.length == 0)
+                options.push("defaults");
```
Comment (cockpituous) [inline pkg/storaged/swap/swap.jsx:47]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240116-090759-3ab49c70-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/swap/swap.jsx [hunk 7]
```diff
@@ -27,14 +27,63 @@
...
+            await block.UpdateConfigurationItem(conf, new_conf, { });
+            return;
+        }
+    }
+
+    await block.AddConfigurationItem(
+        ["fstab", {
+            dir: { t: 'ay', v: encode_filename("none") },
+            type: { t: 'ay', v: encode_filename("swap") },
+            opts: { t: 'ay', v: encode_filename(noauto ? "noauto" : "defaults") },
```
Comment (cockpituous) [inline pkg/storaged/swap/swap.jsx:67]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240116-090759-3ab49c70-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/utils.js [hunk 8]
```diff
@@ -910,6 +911,15 @@ export function get_active_usage(client, path, top_action, child_action, is_temp
                     enter_unmount(children[c], c, false);
                 enter_unmount(block, mp, true);
             });
+        } else if (swap) {
+            if (swap.Active) {
+                usage.push({
+                    level,
+                    usage: 'swap',
+                    block,
+                    actions: get_actions(_("stop")),
```
Comment (cockpituous) [inline pkg/storaged/utils.js:920]: These 7 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240116-090759-3ab49c70-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/utils.js [hunk 9]
```diff
@@ -1024,6 +1034,12 @@
         }
     }
 
+    async function stop_swap(swaps) {
+        for (const s of swaps) {
+            await client.blocks_swap[s.block.path].Stop({});
```
Comment (cockpituous) [inline pkg/storaged/utils.js:1039]: These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240116-090759-3ab49c70-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx (hunk 2 above)
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:432]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240116-114925-80fc371d-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx [hunk 10]
```diff
@@ -450,6 +458,10 @@ function format_dialog_internal(client, path, start, size, enable_dos_extended,
                     partition_type = "21686148-6449-6e6f-744e-656564454649";
                 }
 
+                if (type == "swap") {
+                    partition_type = block_ptable.Type == "dos" ? "0x82" : "0657fd6d-a4ab-43c4-84e5-0933c84b4f4f";
```
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:462]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240116-114925-80fc371d-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx [hunk 11]
```diff
@@ -597,7 +620,10 @@ function format_dialog_internal(client, path, start, size, enable_dos_extended,
                         return (client.wait_for(() => block_fsys_for_block(path))
                                 .then(block_fsys => client.mount_at(client.blocks[block_fsys.path],
                                                                     mount_point)));
-                    if (is_encrypted(vals) && is_filesystem(vals) && !mount_now)
+                    if (type == "swap" && mount_now)
+                        return (client.wait_for(() => client.blocks_swap[path])
+                                .then(block_swap => block_swap.Start({})));
+                    if (is_encrypted(vals) && (is_filesystem(vals) || type == "swap") && !mount_now)
```
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:626]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240116-114925-80fc371d-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/swap/swap.jsx [hunk 12]
```diff
@@ -27,14 +27,63 @@ import { useEvent } from "hooks";
...
```
Comment (cockpituous) [inline pkg/storaged/swap/swap.jsx:67]: These 5 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240116-114925-80fc371d-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/utils.js [hunk 13]
```diff
@@ -910,6 +911,15 @@ export function get_active_usage(client, path, top_action, child_action, is_temp
                     enter_unmount(children[c], c, false);
                 enter_unmount(block, mp, true);
             });
+        } else if (swap) {
```
Comment (cockpituous) [inline pkg/storaged/utils.js:914]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240116-114925-80fc371d-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx (hunk 2 above)
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:432]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240116-134034-00b55033-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx (hunk 10 above)
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:462]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240116-134034-00b55033-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx (hunk 11 above)
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:626]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240116-134034-00b55033-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/swap/swap.jsx (hunk 12 above)
Comment (cockpituous) [inline pkg/storaged/swap/swap.jsx:67]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240116-134034-00b55033-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/utils.js (hunk 13 above)
Comment (cockpituous) [inline pkg/storaged/utils.js:914]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240116-134034-00b55033-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by jelly** [comment]
Some issues I found:

* Creating a swap partition on /dev/sda does not work (no partition table) this should be a valid use case:
```
[root@fedora-39-127-0-0-2-2201 ~]# mkswap /dev/sda
Setting up swapspace version 1, size = 4 GiB (4294963200 bytes)
no label, UUID=934afe18-e205-4bfe-96f2-ec4a76edd783
```

* What about LVM? :)

File: pkg/storaged/block/format-dialog.jsx (hunk 2 above)
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:432]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240117-084217-6aab0478-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx (hunk 3 above)
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:462]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240117-084217-6aab0478-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx [hunk 14]
```diff
@@ -587,6 +610,17 @@
                         return client.blocks_fsys[path];
                 }
 
+                function block_swap_for_block(path) {
+                    if (keep_keys) {
+                        const content_block = client.blocks_cleartext[path];
+                        return client.blocks_swap[content_block.path];
+                    } else if (is_encrypted(vals))
```
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:617]: These 4 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240117-084217-6aab0478-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx [hunk 15]
```diff
@@ -587,6 +610,17 @@
...
+                        return (client.blocks_cleartext[path] &&
+                                client.blocks_swap[client.blocks_cleartext[path].path]);
```
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:619]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240117-084217-6aab0478-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx [hunk 16]
```diff
@@ -597,7 +631,10 @@
                         return (client.wait_for(() => block_fsys_for_block(path))
                                 .then(block_fsys => client.mount_at(client.blocks[block_fsys.path],
                                                                     mount_point)));
-                    if (is_encrypted(vals) && is_filesystem(vals) && !mount_now)
+                    if (type == "swap" && mount_now)
+                        return (client.wait_for(() => block_swap_for_block(path))
+                                .then(block_swap => block_swap.Start({})));
+                    if (is_encrypted(vals) && (is_filesystem(vals) || type == "swap") && !mount_now)
```
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:637]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240117-084217-6aab0478-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/swap/swap.jsx (hunk 12 above)
Comment (cockpituous) [inline pkg/storaged/swap/swap.jsx:67]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240117-084217-6aab0478-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/utils.js (hunk 13 above)
Comment (cockpituous) [inline pkg/storaged/utils.js:914]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240117-084217-6aab0478-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by mvollmer** [comment]
> * Creating a swap partition on /dev/sda does not work (no partition table)

Oh, you mean there is a JavaScript exception? Good catch! Fixed!

**Comment by mvollmer** [comment]
> * Swap on stratis?

That's not possible, stratis only has filesystems, no?

**Comment by jelly** [comment]
> > * Swap on stratis?
> 
> That's not possible, stratis only has filesystems, no?

Ah ok, then nvm!

**Comment by jelly** [comment]
> > * Creating a swap partition on /dev/sda does not work (no partition table)
> 
> Oh, you mean there is a JavaScript exception? Good catch! Fixed!

Yup, blocks_ptable doesn't exist for a plain /dev/sda. Thanks!

File: pkg/storaged/block/format-dialog.jsx (hunk 2 above)
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:432]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240117-111611-70e14048-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx [hunk 17]
```diff
@@ -450,6 +458,12 @@ function format_dialog_internal(client, path, start, size, enable_dos_extended,
                     partition_type = "21686148-6449-6e6f-744e-656564454649";
                 }
 
+                if (type == "swap") {
+                    partition_type = (block_ptable && block_ptable.Type == "dos"
+                        ? "0x82"
```
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:463]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240117-111611-70e14048-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx [hunk 18]
```diff
@@ -587,6 +612,17 @@ function format_dialog_internal(client, path, start, size, enable_dos_extended,
                         return client.blocks_fsys[path];
                 }
 
+                function block_swap_for_block(path) {
+                    if (keep_keys) {
+                        const content_block = client.blocks_cleartext[path];
+                        return client.blocks_swap[content_block.path];
+                    } else if (is_encrypted(vals))
```
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:619]: These 4 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240117-111611-70e14048-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx [hunk 19]
```diff
@@ -587,6 +612,17 @@ function format_dialog_internal(client, path, start, size, enable_dos_extended,
...
+                        return (client.blocks_cleartext[path] &&
+                                client.blocks_swap[client.blocks_cleartext[path].path]);
```
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:621]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240117-111611-70e14048-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx [hunk 20]
```diff
@@ -597,7 +633,10 @@ function format_dialog_internal(client, path, start, size, enable_dos_extended,
                         return (client.wait_for(() => block_fsys_for_block(path))
                                 .then(block_fsys => client.mount_at(client.blocks[block_fsys.path],
                                                                     mount_point)));
-                    if (is_encrypted(vals) && is_filesystem(vals) && !mount_now)
+                    if (type == "swap" && mount_now)
+                        return (client.wait_for(() => block_swap_for_block(path))
+                                .then(block_swap => block_swap.Start({})));
+                    if (is_encrypted(vals) && (is_filesystem(vals) || type == "swap") && !mount_now)
```
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:639]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240117-111611-70e14048-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/resize.jsx [hunk 21]
```diff
@@ -284,6 +284,12 @@ export function get_resize_info(client, block, to_fit) {
                 grow_needs_unmount: false
             };
             shrink_excuse = _("VDO backing devices can not be made smaller");
+        } else if (client.blocks_swap[block.path]) {
```
Comment (cockpituous) [inline pkg/storaged/block/resize.jsx:287]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240117-111611-70e14048-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/swap/swap.jsx (hunk 12 above)
Comment (cockpituous) [inline pkg/storaged/swap/swap.jsx:67]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240117-111611-70e14048-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/utils.js (hunk 13 above)
Comment (cockpituous) [inline pkg/storaged/utils.js:914]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240117-111611-70e14048-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by jelly — DISMISSED** [review]
Looks good, one comment about a potential missing test.

File: test/verify/check-storage-swap [hunk 22]
```diff
@@ -24,27 +24,97 @@ import testlib
...
+    def testBasic(self):
         b = self.browser
+        m = self.machine
+
+        disk = self.add_ram_disk()
 
         self.login_and_go("/storage")
 
-        disk = self.add_ram_disk()
+        # Create a swap partition on GPT
```
[the remark this answers is not in the document]
Comment (mvollmer) [reply test/verify/check-storage-swap:35]: Yes, good point, I am getting lazy...

**Comment by jelly** [comment]
Just realized this is release note worthy so marked it as such.

File: pkg/storaged/block/format-dialog.jsx (hunk 2 above)
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:432]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240118-085557-ffd6b834-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx (hunk 17 above)
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:463]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240118-085557-ffd6b834-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx (hunk 18 above)
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:619]: These 4 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240118-085557-ffd6b834-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx (hunk 19 above)
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:621]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240118-085557-ffd6b834-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx (hunk 20 above)
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:639]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240118-085557-ffd6b834-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/resize.jsx (hunk 21 above)
Comment (cockpituous) [inline pkg/storaged/block/resize.jsx:287]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240118-085557-ffd6b834-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/swap/swap.jsx (hunk 12 above)
Comment (cockpituous) [inline pkg/storaged/swap/swap.jsx:67]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240118-085557-ffd6b834-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/utils.js (hunk 13 above)
Comment (cockpituous) [inline pkg/storaged/utils.js:914]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240118-085557-ffd6b834-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: test/verify/check-storage-swap [hunk 23]
```diff
@@ -24,27 +24,94 @@ import testlib
...
+        self.confirm()
+        b.wait_visible(self.card("Solid State Drive"))
+        m.execute(f"! findmnt --fstab -n -o OPTIONS {disk}1")
+
+        # Format as swap on the command line, starting it should add
+        # fstab entry
+        m.execute(f"mkswap -f {disk}")
+        b.click(self.card_button("Swap", "Start"))
+        b.wait_text(self.card_desc("Swap", "Used"), "0")
+        self.assertIn("defaults", m.execute(f"findmnt --fstab -n -o OPTIONS {disk}"))
```
Comment (jelly) [inline test/verify/check-storage-swap:89]: This check seems to be a bit racy fails on both [Arch](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240118-085557-ffd6b834-arch-storage/log.html#92-1) and [debian-testing](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240118-085557-ffd6b834-debian-testing-storage/log.html#92-1)

**Review by jelly — CHANGES_REQUESTED** [review]
Unfortunately one part of the test seems to be flaky.

File: test/verify/check-storage-swap (hunk 23 above)
Comment (mvollmer) [reply test/verify/check-storage-swap]: Yes, clicking "Start" will first activate the Swap and then add the fstab entry. So it is feasible that the "Used" number appears before the fstab entry. I changed this into a testlib.wait.

File: pkg/storaged/block/format-dialog.jsx (hunk 2 above)
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:432]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240118-104422-e6fc1c3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx (hunk 17 above)
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:463]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240118-104422-e6fc1c3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx (hunk 18 above)
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:619]: These 4 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240118-104422-e6fc1c3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx (hunk 19 above)
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:621]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240118-104422-e6fc1c3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx (hunk 20 above)
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:639]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240118-104422-e6fc1c3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/resize.jsx (hunk 21 above)
Comment (cockpituous) [inline pkg/storaged/block/resize.jsx:287]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240118-104422-e6fc1c3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/swap/swap.jsx (hunk 12 above)
Comment (cockpituous) [inline pkg/storaged/swap/swap.jsx:67]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240118-104422-e6fc1c3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/utils.js (hunk 13 above)
Comment (cockpituous) [inline pkg/storaged/utils.js:914]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240118-104422-e6fc1c3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by jelly — APPROVED** [review]
Approved, re-triggered rhel-8-10-distropkg/storage hopefully it was just a busy test vm.

File: pkg/storaged/block/format-dialog.jsx (hunk 2 above)
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:432]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240118-122938-e6fc1c3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx (hunk 17 above)
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:463]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240118-122938-e6fc1c3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx (hunk 18 above)
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:619]: These 4 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240118-122938-e6fc1c3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx (hunk 19 above)
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:621]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240118-122938-e6fc1c3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx (hunk 20 above)
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:639]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240118-122938-e6fc1c3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/resize.jsx (hunk 21 above)
Comment (cockpituous) [inline pkg/storaged/block/resize.jsx:287]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240118-122938-e6fc1c3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/swap/swap.jsx (hunk 12 above)
Comment (cockpituous) [inline pkg/storaged/swap/swap.jsx:67]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240118-122938-e6fc1c3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/utils.js (hunk 13 above)
Comment (cockpituous) [inline pkg/storaged/utils.js:914]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240118-122938-e6fc1c3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx (hunk 18 above)
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:619]: These 4 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240119-070528-e6fc1c3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/utils.js (hunk 13 above)
Comment (cockpituous) [inline pkg/storaged/utils.js:914]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240119-070528-e6fc1c3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Pull request merged**